### PR TITLE
ibportstate speed/width change support for IBSim

### PR DESCRIPTION
**Pull request by jecavil**

### CODE REVIEW

**Comment by jecavil** [comment]
"Why are these simulator commands needed ? There is ibsendtrap utility in infiniband-diags which can be used for testing sending trap 144." -> This is because we want to change the speed and width through ibportstate.

**Comment by jecavil** [comment]
I can split the pull request into different ones then you can accept what you think is useful for the simulator. I will split it next week.

**Comment by jecavil** [comment]
About the extended speed: Sure, this is currently missing in our changes and this is something we have to add later as well.

**Comment by hnrose** [comment]
Simulator commands aren't needed for that. What am I missing ?

**Comment by jecavil** [comment]
Do you mean the next?
"Speed "nodeid"[port] <new-speed> : change LinkSpeedEnabled"
"Width "nodeid"[port] <new-width> : change LinkWidthEnabled"

We started with this and later we implemented the ibportstate part.
We can remove those commands if you refer to that.

**Comment by hnrose** [comment]
On Thu, Aug 2, 2018 at 10:04 AM, Jesus Camacho Villanueva <
notifications@github.com> wrote:

> Do you mean the next?
> "Speed "nodeid"[port] : change LinkSpeedEnabled"
> "Width "nodeid"[port] : change LinkWidthEnabled"
>
> We started with this and later we implemented the ibportstate part.
> We can remove those commands if you refer to that.
>

Yes, that's (the simulator commands and not the ibportstate part) what I
was referring to.

**Comment by jecavil** [comment]
OK, then we will remove that part :).

File: ibsim/sim_mad.c [hunk 1]
```diff
@@ -540,7 +541,7 @@ do_portinfo(Port * port, unsigned op, uint32_t portnum, uint8_t * data)
 			return ERR_BAD_PARAM;	/* trying to change the state of DOWN port */
 
 		if (p->state == 4) {
-			if (p->lid > 0 && p->lid < maxlinearcap
+			if (node->type != SWITCH_NODE && p->lid > 0 && p->lid < maxlinearcap
```
Comment (hnrose) [inline ibsim/sim_mad.c]: Why is node type != switch needed here ?

Comment (hnrose) [reply ibsim/sim_mad.c]: If this is needed, it should be in a separate patch with more info as to why it's needed/how to reproduce.

I'm going to commit this PR without this change.

**Comment by hnrose** [comment]
Merged commit cfcefe0 (based on this PR) into master

**Comment by jecavil** [comment]
This is fine!

Jesus

**Pull request closed without merging**